### PR TITLE
try to find and link the threads library

### PR DESCRIPTION
**Pull request by adrpo**
### Related Issues

#8242

### Purpose

Find the threads library and link with it for the Cpp/OMSICpp runtime

### CODE REVIEW

**Comment by adrpo** [comment]
The PR was tested on the actual computer (ryzen-5950x) that has the #8242 issue, let's see if it works fine on other systems.

File: OMCompiler/SimulationRuntime/OMSICpp/CMakeLists.txt [hunk 1]
```diff
@@ -479,6 +479,15 @@ IF(FMU_TARGET AND NOT MSVC)
   ADD_DEFINITIONS(-D_GLIBCXX_USE_CXX11_ABI=0)
 ENDIF(FMU_TARGET AND NOT MSVC)
 
+IF(WIN32)
+SET(CPPTHREADS_LIBRARY)
```
[the remark this answers is not in the document]
Comment (adrpo) [reply OMCompiler/SimulationRuntime/OMSICpp/CMakeLists.txt:483]: On Windows it always worked fine, I wanted to restrict the changes to Linux/Mac so we don't break stuff that worked.

[the remark this answers is not in the document]
Comment (sjoelund) [reply OMCompiler/SimulationRuntime/OMSICpp/CMakeLists.txt:483]: Yes, but you removed a pthread below with this variable. Doesn't it remove lining pthread for win32 then?

[the remark this answers is not in the document]
Comment (adrpo) [reply OMCompiler/SimulationRuntime/OMSICpp/CMakeLists.txt:483]: To make sure it works I will add the MinGW label and re-run the PR.

**Pull request merged**